### PR TITLE
Add idiomatic recipes for collection-size assertions.

### PR DESCRIPTION
**Pull request by bbogason**
Closes SEN-2572

### CODE REVIEW

File: recipes/Java/AssertJ/Idiomatic/collection-hasSizeGreaterThan-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        operator: '>'
+              argCount: 1
+              name: assertThat
+              label: assertThatMethodCallIndirect
+availableFixes:
+- name: Use idiomatic assertion
+  actions:
+  - modifyArguments:
+      rewrite:
+        1: '{{{ leftOperand.qualifier }}}'
```
Comment (seanf) [inline recipes/Java/AssertJ/Idiomatic/collection-hasSizeGreaterThan-operator.yaml:60]: I'll bet this was fun the first time. We _really_ need a mustache helper for these.

**Review by seanf — APPROVED** [review]
I just can't review that many YAML recipes in depth after the fact, but they look broadly okay, the IDs seem fine, and the quick-fixes all seem to work as expected.

**Review by wanderneto — APPROVED** [review]
Well, everything seems fine. Recipes are working as expected. Great job!

**Comment by bbogason** [comment]
> I just can't review that many YAML recipes in depth after the fact

Yes, sorry, I got a little carried away - I started with like 4 recipes but kept coming up with minor variations that I felt would also make sense to create recipes for.

It is a bit of a smell that those recipes are so hard to review. Most of the complexity comes because we support both `assertThat(collection).hasSizeGreaterThan(3)` and `assertThat(collection)....as("some helpful message").hasSizeGreaterThan(3)`.

File: recipes/Java/AssertJ/Idiomatic/collection-hasSizeGreaterThan-operator.yaml (hunk 1 above)
Comment (seanf) [reply recipes/Java/AssertJ/Idiomatic/collection-hasSizeGreaterThan-operator.yaml:60]: https://securecodewarrior.atlassian.net/browse/SEN-2501

**Pull request merged**